tvbwidgets/core/hpc/launcher.py

```diff
@@ -190,10 +193,32 @@ def submit_job(self, executable, inputs, do_stage_out):
         else:
             LOGGER.info('You can use "PyUnicore Tasks Stream" tool to monitor it.')
 
+    @staticmethod
+    def read_file_from_hpc(job, file_name):
+        try:
+            content = job.working_dir.listdir()
+            storage_config_file = content.get(file_name)
+            if storage_config_file is None:
+                LOGGER.warning(f"Could not find file: {file_name}")
+                return 0
+            return storage_config_file.raw().read()
+        except Exception as e:
+            LOGGER.error(f"Could not read file: {file_name}", exc_info=e)
+            return 0
+
     def monitor_job(self, job):
         LOGGER.info('Waiting for job to finish...'
                     'It can also be monitored interactively with the "PyUnicore Tasks Stream" tool.')
-        job.poll()
+
+        start_time = int(time.time())
+        # we replaced job.poll to our custom while, to update the progress bar as well
+        while job.status.ordinal() < pyunicore.client.JobStatus.SUCCESSFUL.ordinal():
+            state = int(self.read_file_from_hpc(job, PROGRESS_BAR_STATUS_FILE))
+            self.update_progress(state + 1)
+            time.sleep(2)
+            if self.config.timeout > 0 and int(time.time()) > start_time + self.config.timeout:
+                raise TimeoutError("Timeout waiting for job to become %s" % state.value)
```

Comment:
What happens in the widget when timeout is reached ?
Should we write something in the message area ?